std-bits/bench-processor/src/main/java/org/enso/benchmarks/processor/SpecCollector.java

```diff
@@ -0,0 +1,110 @@
+package org.enso.benchmarks.processor;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.enso.benchmarks.BenchSuite;
+import org.enso.benchmarks.BenchSuiteWrapper;
+import org.enso.polyglot.LanguageInfo;
+import org.enso.polyglot.MethodNames.Module;
+import org.enso.polyglot.MethodNames.TopScope;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.io.IOAccess;
+
+/**
+ * Collect benchmark specifications from source files.
+ */
+public class SpecCollector {
+
+  private final File rootDir;
+  private final Context ctx;
+  private static final String allSuitesVarName = "all";
+  private final String benchPackagePrefix = "local.Benchmarks";
+  private final String ensoSuffix = ".enso";
+
+  public SpecCollector(File benchProjectDir, File languageHomeOverride) {
+    if (!benchProjectDir.exists() || !benchProjectDir.isDirectory()) {
+      throw new IllegalArgumentException(benchProjectDir + " is not a directory.");
+    }
+    this.rootDir = benchProjectDir;
+    this.ctx = Context.newBuilder()
+        .allowExperimentalOptions(true)
+        .allowIO(IOAccess.ALL)
+        .allowAllAccess(true)
+        .logHandler(new ByteArrayOutputStream())
+        .option(
+            "enso.projectRoot",
+            benchProjectDir.getAbsolutePath()
+        )
+        .option(
+            "enso.languageHomeOverride",
+            languageHomeOverride.getAbsolutePath()
+        )
+        .build();
+  }
+
+  public Collection<BenchSuiteWrapper> collectAllBenchSpecs() {
```

Comment:
`SpecCollector` is in `org.enso.benchmarks.processor` because it is coupled with the annotation processing and code generation process, whereas `BenchSuiteWrapper` (that is now renamed to `ModuleBenchSuite`) is in `org.enso.benchmarks` package because it is a *universal* type meant to be used from the generated code.